torchtune/utils/_checkpointing/_checkpointer.py

```diff
@@ -431,6 +439,29 @@ def save_checkpoint(
             dim=self._config["hidden_size"],
         )
 
+        print(f"======={self._model_type}==========")
+        if (
+            self._model_type == "GEMMA"
+        ):  # TODO: Remove this once we have a better way to handle this
+            if (
+                "lm_head.weight" not in self._weight_map.keys()
+                and "lm_head.weight" in state_dict[utils.MODEL_KEY].keys()
+            ):
+                if torch.equal(
+                    state_dict[utils.MODEL_KEY]["lm_head.weight"],
+                    state_dict[utils.MODEL_KEY]["model.embed_tokens.weight"],
+                ):
+                    del state_dict[utils.MODEL_KEY]["lm_head.weight"]
+                    logger.info(
+                        "Delete the model lm_head weight from the state dict "
+                        "because it is the same as the model embed_tokens weight"
+                    )
+                else:
+                    self._weight_map["lm_head.weight"] = "0002"
```

Comment:
When will this else block hit? If we know the checkpoints don't contain this key, let's just work with that assumption? Anyways we're hard coding a bunch of stuff like the name of the key etc.

Reply:
It was because the parameters were not really tied before. Now I removed the else block.